### ISSUE TITLE
Start NVDA after I sign in option is sometimes not preserved

### ISSUE THREAD

**Issue by nvdaes** [nvaccess/nvda#13155]
### Steps to reproduce:
1. Check that Start NVDA after I sign in is checked.
2. Don't restart NVDA with add-ons disabled.
3. Update NVDA to 2021.3
### Actual behavior:
Start NVDA after I sign in is not checked after updating NVDA.
### Expected behavior:
Start NVDA after I sign in should be checked.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2021.3
#### Windows version:
Version 20H2 (built 19042.1348)
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
This doesn't happen when NVDA is installed after restarting with add-ons disabled, but can be reproduced if all add-ons are disabled from add-ons manager, updating from NVDA not restarted with the option --disable-addons.
#### Does the issue still occur after restarting your computer?
Yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes.
[nvda-old.log](https://github.com/nvaccess/nvda/files/7691465/nvda-old.log)

**Comment by lukaszgo1** [comment]
I cannot reproduce this on my Windows 11 machine nor on my Windows 7 box

**Comment by feerrenrut** [comment]
I also can't reproduce this on Windows 10

@nvdaes The steps to reproduce this are not very detailed. If you can reliably reproduce the issue, please take some time to update these instructions to include each step, and collect the log from the installer. Currently the attached log file seems to be for the first run after install.

**Comment by nvdaes** [comment]
Hello, you are right. In fact, steps to reproduce had missing details.
I don't know if this should be closed or fixed in case.

1. Install an add-on which add a section in NVDA configuration, for example, [readFeeds](https://addons.nvda-project.org/files/get.php?file=rf).
2. Set an invalid key (not present in configspec) in the configuration section of the add-on, in nvda.ini. For example: index = 2
3. Check that Start NVDA after I sign in option is checked.
4. Save the configuration.
5. Try to install NVDA.

This doesn't happen if we restart NVDA with add-ons disabled.
[nvdaInstaller.log](https://github.com/nvaccess/nvda/files/7705856/nvdaInstaller.log)

**Comment by lukaszgo1** [comment]
I've managed to reproduce this once following the latest STR, but on an second attempt today it is not reproducible for me any longer. There is definitely something strange going on.

**Comment by CyrilleB79** [comment]
I have no further hint for this issue.
However, I think that this issue has already been seen in the French community mailing list for an earlier upgrade (upgrade to 2021.2 or earlier).

**Comment by seanbudd** [comment]
To investigate this issue it would be be helpful to add logging for raised `WindowsError`s in the following files:

1. `source/config/__init__.py`
1. `source/easeOfAccess.py`

In some instances, such as checking if a key has been set, or deleting a key if it exists, a `WindowsError` `FileNotFoundError` is expected, and therefore an `ERROR` level log would be too noisy.

However, the `FileNotFoundError` could be raised when accessing the [wrong view of the registry (eg 32 vs 64bit)](https://stackoverflow.com/a/36041228); a similar problem was addressed in #12560.

Further investigation and testing is needed to add appropriate logging, that won't add unnecessary noise.

**Comment by seanbudd** [comment]
@nvdaes and others who could reliably reproduce this issue, can you try:
-  reproducing it with [this build](https://ci.appveyor.com/api/buildjobs/6g8i0w5jm273luul/artifacts/output%2Fnvda_snapshot_pr13242-24506%2C86a99510.exe) (based on #13242) with debug logging enabled
- upload the logs for both the installer and the newly installed copy

Hopefully this PR will log an error (with notification) to indicate failure, or at the very least, provide better debugging information.

**Comment by nvdaes** [comment]
Hello. I'll try tomorrow. I get this message when trying to download the built:

{"message":"Artifacts download limit (1024 MB/day) exceeded."}

2022-01-24 6:38 GMT+01:00, Sean Budd ***@***.***>:

**Comment by nvdaes** [comment]
Hello, unfortunately the same message appears when I try to download the built:

{"message":"Artifacts download limit (1024 MB/day) exceeded."}
Hope this helps

2022-01-24 6:38 GMT+01:00, Sean Budd ***@***.***>:

**Comment by seanbudd** [comment]
Thanks @nvdaes  - I've uploaded the launcher to google drive, as github won't let me upload (even within a zip file) https://drive.google.com/file/d/13XW62x59h8YiRxWzVKJJQBl4lThQYvVW/view?usp=sharing

**Comment by nvdaes** [comment]
[nvda-2021.3.1.log](https://github.com/nvaccess/nvda/files/7947852/nvda-2021.3.1.log)
[nvda-pr13242.log](https://github.com/nvaccess/nvda/files/7947853/nvda-pr13242.log)
Hello, I've reproduced the issue and here are logs. Hope this helps.

**Comment by seanbudd** [comment]
Apologies @nvdaes, I gave incorrect instructions. We need to use PR 13242 as both the base, and the installer and capture 3 logs to fully capture where the problem could occur.

Download these 2 installers: https://drive.google.com/file/d/11fLmw4i9Bv5c4NYhiIuLID2cHUNWNWvG/view?usp=sharing

1. install an older build of PR 13242: `nvda_snapshot_pr13242-24506,86a99510.exe`
1. turn on "start after logon", save the config
1. Capture the NVDA log
1. install a newer build of PR 13242: `nvda_snapshot_pr13242-24544,202c3758.exe`, to cause an update to occur
1. capture the installer log
1. start the newly installed copy
1. open the settings, check if "start after logon" is enabled
1. capture the latest NVDA log
1. if "start after logon" is no longer enabled, upload all 3 logs

**Comment by nvdaes** [comment]
Hello, now the start after logon option is properly preserved
(checked). Thanks for fixing this! 👌

2022-01-28 2:59 GMT+01:00, Sean Budd ***@***.***>:

**Comment by seanbudd** [comment]
Is this consistently fixed? I was not expecting this PR to fix anything.

**Comment by nvdaes** [comment]
With the previous instructions the option was not preserved and start
after logon was disabled after installing the snapshot. Now it's
checked. Should I test anything else?

2022-01-28 5:53 GMT+01:00, Sean Budd ***@***.***>:

**Comment by seanbudd** [comment]
If it is still possible to reproduce the loss of "start after logon", you can perform the steps again to reproduce the issue and forward us the logs.
Otherwise, we can merge #13242 and try to capture the same logs in alpha updates if the issue persists.

**Comment by nvdaes** [comment]
Hello, I'll try again after job, in a few hours. I'll try to install
the new version of NVDA since downgrading NVDA is not recommended. But
seems that the problem is fixed anyway.

2022-01-28 6:42 GMT+01:00, Sean Budd ***@***.***>:

**Comment by lukaszgo1** [comment]
It is only an educated guess, but perhaps the "start after logon" fails to be preserved only when signed builds are used. @seanbudd  Would it be possible to create a try branch from the code in PR #13242 so that builds @nvdaes  uses for testing are as close to the official releases as possible?

**Comment by nvdaes** [comment]
[nvda4.zip](https://github.com/nvaccess/nvda/files/7959315/nvda4.zip)
Hello: the issue persists. I've just installed the last provided version for testing. Here are the 4 logs.

**Comment by CyrilleB79** [comment]
This issue seems to be the same as #8720. One of them should probably be closed as a duplicate.

**Comment by seanbudd** [comment]
Closing, see https://github.com/nvaccess/nvda/issues/8720#issuecomment-1082605799 for a summary of this issue
